lib/spell/message.ex

```diff
@@ -218,13 +218,17 @@ defmodule Spell.Message do
 
   @doc """
   Return a new WAMP id.
+
+  To ensure the uniqueness of the new id we use :crypto.rand_bytes to generate
+  a random seed
+
+  TODO: improve `:random.uniform` using a Mersenne Twister PRNG algorithm
   """
   @spec new_id :: integer
   def new_id do
-    # TODO: hack to ensure the random generator is seeded -- will result in
-    # clock skew
-    :random.seed(:erlang.now())
-    (:math.pow(2, 53) |> round |> :random.uniform) - 1
+    << a :: 32, b :: 32, c :: 32 >> = :crypto.rand_bytes(12)
+    :random.seed(a,b,c)
+    :math.pow(2, 53) |> round |> :random.uniform
```

Comment:
Yup, it is as @danmarcab says --

> uniform/1 returns a random integer uniformly distributed between 1 and N

`- 1` changes the range from 0 to N - 1. But! this isn't quite right -- from the basic WAMP spec, the id should be

> between (inclusive) 0 and 2^53 (9007199254740992)

Given that, I think the line should be:

```elixir
:random.uniform(:math.pow(2, 53) + 1) - 1
```

That `round` call can be cut -- it's cruft from using `uniform/0`.